clang/include/clang/Lex/Preprocessor.h

```diff
@@ -2304,7 +2304,8 @@ class Preprocessor {
 
   /// Check whether the next pp-token is one of the specificed token kind. this
   /// method should have no observable side-effect on the lexed tokens.
-  template <tok::TokenKind K, tok::TokenKind... Ks> bool isNextPPTokenOneOf() {
+  template <typename... Ts>
+  bool isNextPPTokenOneOf(tok::TokenKind K, Ts... Ks) {
```

Comment:
Could we make it:

```suggestion
  bool isNextPPTokenOneOf(Ts... Ks) {
```

?

Though I guess an empty pack here ends up going through all of the other work, which is... awkward.  But a `static_assert(sizeof(Ks) > 0)` accomplishes that too?  

IDK, something to think about.

Reply:
Yes, I think `bool isNextPPTokenOneOf(Ts... Ks) with static_assert(sizeof...(Ks))` is equal to `bool isNextPPTokenOneOf(tok::TokenKind K, Ts... Ks)`. But static_assert may give a more clear diagnostic. WDYT?

Reply:
Updated to `bool isNextPPTokenOneOf(Ts... Ks) with static_assert(sizeof...(Ks))` version.

Reply:
@erichkeane @zwuis  Can I land this patch?